### PR TITLE
Refactor compliant filters with elastic date ranges

### PR DESCRIPTION
**Pull request by pablopunk**
Not a task, just something that was missing on a recent refactor

### CODE REVIEW

File: scripts/apps/search/directives/DateFilters.ts [hunk 1]
```diff
@@ -7,6 +7,24 @@ interface IDateRange {
...
+    },
+};
+
+const next3MonthsFilter: IDateRange = {
+    key: 'next_3_months',
+    label: gettext('Next 3 Months'),
+    elasticSearchDateRange: {
+        lte: 'now+3M/M',
+        gte: 'now+3M/M',
+    },
```
Comment (tomaskikutis) [inline scripts/apps/search/directives/DateFilters.ts]: If today is 2nd Oct, should next 3 months mean 2nd October to 31st December inclusive or November 1st - January 1st inclusive or something different?

Comment (pablopunk) [reply scripts/apps/search/directives/DateFilters.ts]: it should be from today (Oct 2nd) to January 2nd inclusive

Comment (petrjasek) [reply scripts/apps/search/directives/DateFilters.ts]: I think you should round it to day then `/d` otherwise it would be till the end of month `/M`

Comment (pablopunk) [reply scripts/apps/search/directives/DateFilters.ts]: You're right. I'm also thinking I need to remove `gte` as we also need old articles to be marked as "overdue"

File: scripts/apps/search/directives/DateFilters.ts [hunk 2]
```diff
@@ -51,6 +67,8 @@ const last8hoursFilter: IDateRange = {
 };
 
 export const dateRangesByKey: Dictionary<string, IDateRange> = {
+    before_1_month: before1MonthFilter,
```
Comment (tomaskikutis) [inline scripts/apps/search/directives/DateFilters.ts]: `before_next_month` and `before_3_months_ahead` sound better to me.

**Pull request merged**